dulwich/index.py

```diff
@@ -625,23 +651,21 @@ def get_unstaged_changes(index, root_path, filter_blob_callback=None):
             if filter_blob_callback is not None:
                 blob = filter_blob_callback(blob, tree_path)
         except OSError as e:
```

Comment:
Can't we just change OSError here to (IOError, OSError)?

Reply:
I have updated the code, thanks for the idea